src/components/NewCommentForm.tsx

```diff
@@ -1,99 +1,119 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import { addComment } from '../api';
+import { GlobalContext } from '../reducer';
+import { Comment } from '../types/Comment';
+import { Error } from '../types/Error';
+import { Field } from './Field';
 
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
+type Props = {
+  getComments: (id:number) => void
+};
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
+export const NewCommentForm: React.FC<Props> = ({ getComments }) => {
+  const [state, dispatch] = useContext(GlobalContext);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [valideFrom, setValidForm] = useState(
+    { name: true, email: true, text: true },
+  );
 
-          <span className="icon is-small is-left">
-            <i className="fas fa-user" />
-          </span>
+  const clearForm = () => {
+    setName('');
+    setEmail('');
+    setText('');
+    setValidForm({ name: true, email: true, text: true });
+  };
 
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
-        </div>
+  const addCommentSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const valid = { ...valideFrom };
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
-      </div>
+    if (!name) {
+      valid.name = false;
+    }
 
-      <div className="field" data-cy="EmailField">
-        <label className="label" htmlFor="comment-author-email">
-          Author Email
-        </label>
+    if (!email) {
+      valid.email = false;
+    }
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="email"
-            id="comment-author-email"
-            placeholder="email@test.com"
-            className="input is-danger"
-          />
+    if (!text) {
+      valid.text = false;
+    } else if (state.selectedPost && state.selectedUser) {
```

Comment:
Why you need check above? You requst will be done without name or email as I undestand. best way to validate - it's each field onChange

Reply:
yes i push new code, but if i won't check state.selectPost typescript will be swear, because state.selectedPost can be null, i wrote so in reduce